### PR TITLE
Add rpmbuild container call

### PR DESCRIPTION
**Pull request by scoheb**
- replace duffy calls with container step for rpmbuild stage
- refactor to allow pipeline to be parameterized for provisioning

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -139,48 +151,20 @@ podTemplate(name: 'fedora-atomic-inline', label: 'fedora-atomic-inline', cloud:
...
-                        writeFile file: "${env.ORIGIN_WORKSPACE}/task.env",
-                                text: "export JENKINS_JOB_NAME=\"${JOB_NAME}-${current_stage}\"\n" +
-                                        "export JENKINS_BUILD_TAG=\"${BUILD_TAG}-${current_stage}\"\n" +
-                                        "export OSTREE_BRANCH=\"${OSTREE_BRANCH}\"\n" +
-                                        "export fed_repo=\"${fed_repo}\"\n" +
-                                        "export fed_branch=\"${fed_branch}\"\n" +
-                                        "export fed_rev=\"${fed_rev}\"\n"
-                        rsyncResults("${current_stage}")
-
-                        def package_props = "${env.ORIGIN_WORKSPACE}/logs/package_props.txt"
```
Comment (p3ck) [inline Jenkinsfile]: I assume there are some properties from the rpm build container that need to be exposed to the next stage?  How does that work in this model?

Comment (scoheb) [inline Jenkinsfile]: nice catch. Yes, I presume these props need to be passed...I will add them back

File: Jenkinsfile [hunk 2]
```diff
@@ -139,48 +151,25 @@ podTemplate(name: 'fedora-atomic-inline', label: 'fedora-atomic-inline', cloud:
...
-
-                        if (env.OSTREE_BRANCH == null) {
-                            env.OSTREE_BRANCH = ""
+                        // Call rpmbuild-test.sh
+                        // in rpmbuild container
+                        container('rpmbuild') {
+                            sh "fed_repo=${env.fed_repo} " +
+                               "fed_branch=${env.fed_branch} " +
+                               "fed_rev=${env.fed_rev} " +
+                               "/home/rpmbuild-test.sh"
```
Comment (johnbieren) [inline Jenkinsfile]: We will also need RSYNC_PASSWORD, enable_rsync=yes and production=yes (my apologies if they are in here somewhere else and I missed them)

**Comment by arilivigni** [comment]
Can we not make this change until I have the libs in place otherwise I have to keep playing catchup.  I will have an initial review shortly and we can make changes but I don't think this is a blocker is it?

**Comment by scoheb** [comment]
For sure...we can wait...we actually need the staging env first before we merge this I would think too...

**Comment by arilivigni** [comment]
Sounds good thanks!

**Comment by scoheb** [comment]
Closed in favor of https://github.com/CentOS-PaaS-SIG/ci-pipeline/pull/348

**Pull request closed without merging**